### ISSUE TITLE
"Unique" field attribute & "Safe" sails' migrate option

### ISSUE THREAD

**Issue by ParanormalCoder** [balderdashy/waterline#1246]
I'm not sure if it's a bug. My sails config in package.json is:

    "sails": "~0.12.0-rc3",
    "sails-mongo": "^0.11.5"

So, this version of sails contains waterline version 0.10.28.

There are two cases:
1- To set the migrate:"safe" inside the models.js in project's config folder
2- To enter number "1"/enter-key at the beginning of the sails app load & run

In both above cases, the unique attribute inside the models is ignored. (for example having a model User, and running this command: User.create({username:"repeated'})...)

If it's a bug, just help us to resolve it. If it's a normal thing for the sails project, please document it somewhere.

Thanks

**Comment by ParanormalCoder** [comment]
Ok, my summary from testing development & production next to different migrate options.
It seems if the db initially is made by the sails in "safe" migrate mode, the unique fields are ignored.

So the solution I found:
Initially run the app in "drop" or "alter" migrate mode, then start to use "safe" migrate mode.

**Comment by devinivy** [comment]
Correct– migrations are not run on production, which essentially forces `migrate: safe`.  They should be done using a different tool.  Perhaps unique indexes could be created when using the `create` migration strategy.

**Comment by sailsbot** [comment]
Thanks for posting, @ParanormalCoder.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
